### PR TITLE
feat(listings integration): Integrate Frontend Listings Fragment with Backend

### PR DESCRIPTION
**Pull request by MphiGithub**
# Welcome to VanRoomies! 👋✈️

Fixes: #39

## Description of the change:
Integrates the frontend listings page with the backend. Currently, the page takes listing titles and the first image.
Due to limitations with .toString(), images larger than 8x8 (in base 64 format) are not accepted and will crash the app.
This will be looked into with issue #47.

## How to manually test:
1. Start up MongoDB
2. Run `npm run dev`
3. Go to Postman and run `POST Create New User`
4. Take the outputted `_id` and run `POST Create Listing under User (With Image)` with the id as `userId`
5. Repeat step 4 as many times as you'd like
6. Take `_id` and paste it into `ListingsFragment` near the end of `onCreateView`. The userId is hardcoded at the moment. 
image
7. Build and run the app
8. The listings should show up in the listings menu!

### CODE REVIEW

File: frontend/app/src/main/java/com/chads/vanroomies/ListingsRecyclerData.java [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // Reference: https://www.geeksforgeeks.org/recyclerview-using-gridlayoutmanager-in-android-with-example/
 public class ListingsRecyclerData {
     private String title;
-    private int imgId;
+    private String imageString;
```
Comment (FarihaIS) [inline frontend/app/src/main/java/com/chads/vanroomies/ListingsRecyclerData.java:5]: change to imageId?

**Review by FarihaIS — CHANGES_REQUESTED** [review]
PR functional, just some formatting stuff to review!

File: frontend/app/src/main/java/com/chads/vanroomies/Constants.java [hunk 2]
```diff
@@ -2,7 +2,10 @@
 
 public class Constants {
     // Endpoints
-    final static String localBaseServerURL = "https://10.0.2.2:3000";
+    final static String BaseServerURL = "https://10.0.2.2:3000";
```
[the remark this answers is not in the document]
Comment (MphiGithub) [reply frontend/app/src/main/java/com/chads/vanroomies/Constants.java:5]: Oops! Dunno how I missed that, thanks! I'll propagate this change into my other PRs that use baseServerURL after this is merged!

File: frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java [hunk 3]
```diff
@@ -1,18 +1,38 @@
 package com.chads.vanroomies;
 
+import android.app.Activity;
```
[the remark this answers is not in the document]
Comment (MphiGithub) [reply frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java:3]: Yep!

File: frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java [hunk 4]
```diff
@@ -1,18 +1,38 @@
...
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
 import android.os.Bundle;
```
[the remark this answers is not in the document]
Comment (MphiGithub) [reply frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java:7]: Will do! Think those were automatic

File: frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java [hunk 5]
```diff
@@ -21,6 +41,8 @@
  */
 public class ListingsFragment extends Fragment {
     final static String TAG = "ListingsFragment";
+    OkHttpClient client;
+    final static Gson g = new Gson();
     final static int num_cols = 2;
```
[the remark this answers is not in the document]
Comment (MphiGithub) [reply frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java:46]: Sure!

File: frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java [hunk 6]
```diff
@@ -21,6 +41,8 @@
  */
 public class ListingsFragment extends Fragment {
     final static String TAG = "ListingsFragment";
+    OkHttpClient client;
```
[the remark this answers is not in the document]
Comment (MphiGithub) [reply frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java:44]: Sure

File: frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java [hunk 7]
```diff
@@ -69,27 +91,56 @@ public void onCreate(Bundle savedInstanceState) {
...
+                Log.d(TAG, e.getMessage());
+            }
+
+            @Override
+            public void onResponse(@NonNull Call call, @NonNull Response response) {
+                act.runOnUiThread(() -> {
+                    try {
+                        recyclerDataArrayList = new ArrayList<>();
+                        String responseData = response.body().string();
+                        List<Map<String, Object>> result = g.fromJson(responseData, List.class);
```
[the remark this answers is not in the document]
Comment (MphiGithub) [reply frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java:117]: sure, I'll call it responseDataList

File: frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java [hunk 8]
```diff
@@ -69,27 +91,56 @@ public void onCreate(Bundle savedInstanceState) {
...
+
+                        for (int index = 0; index < result.size(); index++){
+                            Map<String, Object> listing_json = result.get(index);
```
[the remark this answers is not in the document]
Comment (MphiGithub) [reply frontend/app/src/main/java/com/chads/vanroomies/ListingsFragment.java:120]: on it

File: frontend/app/src/main/java/com/chads/vanroomies/ListingsRecyclerData.java (hunk 1 above)
Comment (MphiGithub) [reply frontend/app/src/main/java/com/chads/vanroomies/ListingsRecyclerData.java:5]: imsageString contains it is an image as a Base 64 encoded String, so I think it should be kept this way (not an id)

**Review by FarihaIS — APPROVED** [review]
Thanks for resolving comments!

**Pull request merged**